lib/sles4sap_publiccloud.pm

```diff
@@ -1393,6 +1393,33 @@ sub wait_for_zypper {
     die("Zypper is still locked after $args{max_retries} retries, aborting (rc: 7)") if $retry >= $args{max_retries};
 }
 
+sub check_zypper_ref {
+    my ($self, %args) = @_;
+    croak('Argument <instance> missing') unless $args{instance};
+    $args{runas} //= 'cloudadmin';
+
+    my $ret = $self{my_instance}->run_ssh_command(cmd => 'sudo zypper ref',
+        username => $args{runas},
+        proceed_on_failure => 1,
+        rc_only => 1,
+        quiet => 1,
+        timeout => $args{timeout});
+    switch($ret) {
```

Comment:
I think this will need a `use Switch;`. See: https://metacpan.org/pod/Switch

Is it working as is?